Model Submission for DS1 sensor configuration 1 from Explorer

HongbiaoZ opened the pull request.

acschang commented:
Thank you for robot model submission however we are unable to evaluate your submission until your submission meets the criteria laid out in the [Simulation Model Preparation Guide](https://www.subtchallenge.com/resources/Simulation_Model_Preparation_Guide.pdf).  Please add a `specifications.md` file to your submission with details laid out in the aforementioned guide.

HongbiaoZ commented:
Thank you. I have added a specifications.md to the models.

acschang commented:
The initial assessment of the Explorer R2 model is complete, however there are some issues to resolve before this model can be merged and utilized in a competitive setting:

- The mass listed in `specifications.md` does not match the mass in `model.sdf`.  Please ensure both files reflect the physical robot's mass.
- The model does not turn when yaw commands are applied on the `cmd_vel` topic. Please check and correct the vehicle's ability to achieve the commanded yaw velocity. We will work to help resolve the `diff_drive` implementation.
- Any limits on the physical robot's rotational suspension (maximum angles or non-zero spring coefficients) should be reflected in the `Center Pivot` joint in `model.sdf` and noted in `specifications.md`.
- The second IMU listed in the `specifications.md` file is not present in the model. Please rectify this issue by adding the additional IMU sensor at the proper location in the model.
- The validation data is necessary to complete the evaluation. If not provided, the model's maximum velocity will be limited to 1m/s and battery life set to 60min. When the model validation data can be provided later, please submit a PR updating the model.

Please address the issues noted above and then submit a pull request to this new branch to update your submission by 15 JULY. The submissions are expected to be incorporated into the SubT Virtual Testbed pending a successful review.

acschang commented:
The initial assessment of the Explorer DS1 model is complete, however there are some issues to resolve before this model can be merged and utilized in a competitive setting:

- The mass listed in `specifications.md` does not match the mass in `model.sdf`. Please ensure both files reflect the physical robot's mass.
- The vehicle does not currently possess rotors. Please add spinning rotor components.
- The light sources do not have visual mesh components and are not affixed to the platform.  Please add visual components for each light source and affix them to the model based upon the physical platform's mounted lights.
- The model's motion parameters match other standard UAVs in the SubT Tech Repo so it will not be degraded further, but the validation data is necessary to complete the evaluation. When the model validation data can be provided later, please submit a PR updating the model.

Please address the issues noted above and then submit a pull request to [this new branch](https://github.com/osrf/subt/pull/483) to update your submission by **15 JULY**. The submissions are expected to be incorporated into the SubT Virtual Testbed pending a successful review.

HongbiaoZ commented:
Hi acschang,

I have modified the packages according to the issues you sent me and pushed
them to submitted_models/explorer_r2_sensor_config1
<https://github.com/osrf/subt/tree/submitted_models/explorer_r2_sensor_config1>
branch on osrf/subt <https://github.com/osrf/subt>. For the forth issue
with imu, I changed the number of imu to 1 because we only use one IMU in
the competition although we have two. For the last issue, we will work on
the validation datas  right now. When we finish, I will submit a new pull
request.
Thank you.

Regards.

On Fri, 10 Jul 2020 at 15:02, acschang <notifications@github.com> wrote:

> The initial assessment of the Explorer R2 model is complete, however there
> are some issues to resolve before this model can be merged and utilized in
> a competitive setting:
>
>    - The mass listed in specifications.md does not match the mass in
>    model.sdf. Please ensure both files reflect the physical robot's mass.
>    - The model does not turn when yaw commands are applied on the cmd_vel
>    topic. Please check and correct the vehicle's ability to achieve the
>    commanded yaw velocity. We will work to help resolve the diff_drive
>    implementation.
>    - Any limits on the physical robot's rotational suspension (maximum
>    angles or non-zero spring coefficients) should be reflected in the Center
>    Pivot joint in model.sdf and noted in specifications.md.
>    - The second IMU listed in the specifications.md file is not present
>    in the model. Please rectify this issue by adding the additional IMU sensor
>    at the proper location in the model.
>    - The validation data is necessary to complete the evaluation. If not
>    provided, the model's maximum velocity will be limited to 1m/s and battery
>    life set to 60min. When the model validation data can be provided later,
>    please submit a PR updating the model.
>
> Please address the issues noted above and then submit a pull request to
> this new branch to update your submission by 15 JULY. The submissions are
> expected to be incorporated into the SubT Virtual Testbed pending a
> successful review.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/osrf/subt/pull/451#issuecomment-656836116>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AO2OAXRTFZQLSTLSZK6PJGTR25QV3ANCNFSM4NX67E4A>
> .
>


-- 
Hongbiao Zhu
The Field Robotics Center
Robotics Institution
Carnegie Mellon Univercity

HongbiaoZ commented:
We are working on DS1 model now. Hopefully, we can provide validation data
for both ds1 and r2 this week. Thank you for all your work.

On Fri, 10 Jul 2020 at 17:29, acschang <notifications@github.com> wrote:

> The initial assessment of the Explorer DS1 model is complete, however
> there are some issues to resolve before this model can be merged and
> utilized in a competitive setting:
>
>    - The mass listed in specifications.md does not match the mass in
>    model.sdf. Please ensure both files reflect the physical robot's mass.
>    - The vehicle does not currently possess rotors. Please add spinning
>    rotor components.
>    - The light sources do not have visual mesh components and are not
>    affixed to the platform. Please add visual components for each light source
>    and affix them to the model based upon the physical platform's mounted
>    lights.
>    - The model's motion parameters match other standard UAVs in the SubT
>    Tech Repo so it will not be degraded further, but the validation data is
>    necessary to complete the evaluation. When the model validation data can be
>    provided later, please submit a PR updating the model.
>
> Please address the issues noted above and then submit a pull request to this
> new branch <https://github.com/osrf/subt/pull/483> to update your
> submission by *15 JULY*. The submissions are expected to be incorporated
> into the SubT Virtual Testbed pending a successful review.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/osrf/subt/pull/451#issuecomment-656899742>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AO2OAXXWGMRFIX6JOPPI4L3R26B35ANCNFSM4NX67E4A>
> .
>


-- 
Hongbiao Zhu
The Field Robotics Center
Robotics Institution
Carnegie Mellon Univercity

nkoenig commented:
This PR has been superseded by #483.
